File: app/src/main/java/com/sda/books/reader/entity/UpdateDio.java
package com.sda.books.reader.entity;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class UpdateDio {

    @SerializedName("platform")
    private String platform;
    @SerializedName("channel")
    private String channel;
    @SerializedName("current_version")
    private String currentVersion;
    @SerializedName("latest_version")
    private String latestVersion;
    @SerializedName("update_available")
    private Boolean updateAvailable;
    @SerializedName("mandatory")
    private Boolean mandatory;
    @SerializedName("assets")
    private AssetsDTO assets;
    @SerializedName("release_notes")
    private List<String> releaseNotes;
    @SerializedName("released_at")
    private String releasedAt;
    @SerializedName("server_time")
    private String serverTime;

    public String getPlatform() {
        return platform;
    }

    public void setPlatform(String platform) {
        this.platform = platform;
    }

    public String getChannel() {
        return channel;
    }

    public void setChannel(String channel) {
        this.channel = channel;
    }

    public String getCurrentVersion() {
        return currentVersion;
    }

    public void setCurrentVersion(String currentVersion) {
        this.currentVersion = currentVersion;
    }

    public String getLatestVersion() {
        return latestVersion;
    }

    public void setLatestVersion(String latestVersion) {
        this.latestVersion = latestVersion;
    }

    public Boolean getUpdateAvailable() {
        return updateAvailable;
    }

    public void setUpdateAvailable(Boolean updateAvailable) {
        this.updateAvailable = updateAvailable;
    }

    public Boolean getMandatory() {
        return mandatory;
    }

    public void setMandatory(Boolean mandatory) {
        this.mandatory = mandatory;
    }

    public AssetsDTO getAssets() {
        return assets;
    }

    public void setAssets(AssetsDTO assets) {
        this.assets = assets;
    }

    public List<String> getReleaseNotes() {
        return releaseNotes;
    }

    public void setReleaseNotes(List<String> releaseNotes) {
        this.releaseNotes = releaseNotes;
    }

    public String getReleasedAt() {
        return releasedAt;
    }

    public void setReleasedAt(String releasedAt) {
        this.releasedAt = releasedAt;
    }

    public String getServerTime() {
        return serverTime;
    }

    public void setServerTime(String serverTime) {
        this.serverTime = serverTime;
    }

    public static class AssetsDTO {
        @SerializedName("url")
        private String url;
        @SerializedName("size")
        private String size;
        @SerializedName("sha256")
        private String sha256;

        public String getUrl() {
            return url;
        }

        public void setUrl(String url) {
            this.url = url;
        }

        public String getSize() {
            return size;
        }

        public void setSize(String size) {
            this.size = size;
        }

        public String getSha256() {
            return sha256;
        }

        public void setSha256(String sha256) {
            this.sha256 = sha256;
        }
    }
}

File: app/src/main/java/com/sda/books/reader/entity/UpdateDio.kt
package com.sda.books.reader.entity

import com.google.gson.annotations.SerializedName

/**
 * 更新信息数据类
 * 使用 Kotlin data class 提供更安全的字段设计和空值处理
 */
data class UpdateDio(
    @SerializedName("platform")
    val platform: String? = null,
    
    @SerializedName("channel")
    val channel: String? = null,
    
    @SerializedName("current_version")
    val currentVersion: String? = null,
    
    @SerializedName("latest_version")
    val latestVersion: String? = null,
    
    @SerializedName("update_available")
    val updateAvailable: Boolean? = null,
    
    @SerializedName("mandatory")
    val mandatory: Boolean? = null,
    
    @SerializedName("assets")
    val assets: Assets? = null,
    
    @SerializedName("release_notes")
    val releaseNotes: List<String>? = null,
    
    @SerializedName("released_at")
    val releasedAt: String? = null,
    
    @SerializedName("server_time")
    val serverTime: String? = null
) {
    /**
     * 检查是否有更新可用（安全访问）
     */
    fun hasUpdate(): Boolean = updateAvailable == true
    
    /**
     * 检查是否为强制更新（安全访问）
     */
    fun isMandatory(): Boolean = mandatory == true
    
    /**
     * 获取下载 URL（安全访问）
     */
    fun getDownloadUrl(): String? = assets?.url
    
    /**
     * 获取文件大小（安全访问）
     */
    fun getFileSize(): String? = assets?.size
    
    /**
     * 获取 SHA256 校验值（安全访问）
     */
    fun getSha256(): String? = assets?.sha256
    
    /**
     * 获取发布说明（安全访问，返回空列表而不是 null）
     */
    fun getReleaseNotesSafe(): List<String> = releaseNotes ?: emptyList()
    
    /**
     * 获取发布说明文本（安全访问）
     */
    fun getReleaseNotesText(): String = getReleaseNotesSafe().joinToString("\n")
    
    /**
     * 资源信息数据类
     */
    data class Assets(
        @SerializedName("url")
        val url: String? = null,
        
        @SerializedName("size")
        val size: String? = null,
        
        @SerializedName("sha256")
        val sha256: String? = null
    ) {
        /**
         * 检查资源信息是否完整
         */
        fun isValid(): Boolean = !url.isNullOrBlank() && !size.isNullOrBlank()
    }
}
